src/main/java/de/hysky/skyblocker/skyblock/item/tooltip/AccessoriesHelper.java

```diff
@@ -0,0 +1,230 @@
+				if (ACCESSORY_BAG_TITLE.matcher(genericContainerScreen.getTitle().getString()).matches()) {
+					ScreenEvents.afterRender(screen).register((_screen, _context, _mouseX, _mouseY, _delta) -> {
+						GenericContainerScreenHandler handler = genericContainerScreen.getScreenHandler();
+
+						collectAccessories(handler.slots.subList(0, handler.getRows() * 9));
+					});
+				}
+			}
+		});
+	}
+
+	private static void load() {
+		loaded = CompletableFuture.runAsync(() -> {
+			try (BufferedReader reader = Files.newBufferedReader(FILE)) {
+				COLLECTED_ACCESSORIES.putAll(ProfileAccessoryData.SERIALIZATION_CODEC.parse(JsonOps.INSTANCE, JsonParser.parseReader(reader)).result().orElseThrow());
+			} catch (NoSuchFileException ignored) {
+			} catch (Exception e) {
+				LOGGER.error("[Skyblocker Accessory Helper] Failed to load accessory file!", e);
+			}
+		});
+	}
+
+	private static void save() {
+		try (BufferedWriter writer = Files.newBufferedWriter(FILE)) {
+			SkyblockerMod.GSON.toJson(ProfileAccessoryData.SERIALIZATION_CODEC.encodeStart(JsonOps.INSTANCE, COLLECTED_ACCESSORIES).result().orElseThrow(), writer);
+		} catch (Exception e) {
+			LOGGER.error("[Skyblocker Accessory Helper] Failed to save accessory file!", e);
+		}
+	}
+
+	private static void collectAccessories(List<Slot> slots) {	
+		//Is this even needed?
+		if (!loaded.isDone()) return;
+
+		List<String> accessoryIds = slots.stream()
+				.map(Slot::getStack)
+				.map(ItemUtils::getItemId)
+				.filter(NON_EMPTY)
+				.collect(Collectors.toUnmodifiableList());
+
+		String uuid = UndashedUuid.toString(MinecraftClient.getInstance().getSession().getUuidOrNull());
+
+		Map<String, ProfileAccessoryData> playerData = COLLECTED_ACCESSORIES.computeIfAbsent(uuid, _uuid -> new Object2ObjectOpenHashMap<>());
+		playerData.putIfAbsent(Utils.getProfileId(), ProfileAccessoryData.createDefault());
+
+		ProfileAccessoryData profileData = playerData.get(Utils.getProfileId());
+
+		profileData.accessoryIds().addAll(accessoryIds);
+	}
+
+	static AccessoryReport calculateReport4Accessory(String accessoryId) {
+		if (!ACCESSORY_DATA.containsKey(accessoryId) || Utils.getProfileId().isEmpty()) return AccessoryReport.INELIGIBLE;
+
+		Accessory accessory = ACCESSORY_DATA.get(accessoryId);		
+		String uuid = UndashedUuid.toString(MinecraftClient.getInstance().getSession().getUuidOrNull());
+		Set<Accessory> collectedAccessories = COLLECTED_ACCESSORIES.get(uuid).get(Utils.getProfileId()).accessoryIds().stream()
+				.filter(ACCESSORY_DATA::containsKey)
+				.map(ACCESSORY_DATA::get)
+				.collect(Collectors.toSet());
+
+		//If the player has this accessory and it doesn't belong to a family
+		if (collectedAccessories.contains(accessory) && accessory.family().isEmpty()) return AccessoryReport.HAS_HIGHEST_TIER;
+
+		Predicate<Accessory> HAS_SAME_FAMILY = accessory::hasSameFamily;
```

Comment:
Just inline this?

Reply:
Again reduces memory usage as this is needed twice.